app/src/main/java/org/schabi/newpipe/util/Localization.java

```diff
@@ -78,10 +79,13 @@ public static String concatenateStrings(final String... strings) {
         return concatenateStrings(DOT_SEPARATOR, Arrays.asList(strings));
     }
 
+    // Use of BidiFormater to fix text direction automatically
     @NonNull
     public static String concatenateStrings(final String delimiter, final List<String> strings) {
+        final BidiFormatter bidi = BidiFormatter.getInstance();
         return strings.stream()
                 .filter(string -> !TextUtils.isEmpty(string))
+                .map((s) -> strings.indexOf(s) == 0 ? bidi.unicodeWrap(s) : s) // only wrap username
```

Comment:
Hm, I’m not sure we should add this logic to `concatenateStrings`, if you look at the call graph this function gets used in a lot of places, not just the username logic:
![image](https://github.com/user-attachments/assets/15b930ed-1d8a-4ed9-8d74-9646493f1aad)